dashboard/src/shared/Kube.test.ts

```diff
@@ -53,8 +84,19 @@ describe("App", () => {
         data: resource,
       });
       expect(moxios.requests.mostRecent().url).toBe(
-        `${KUBE_ROOT_URL}/api/v1/namespaces/default/pods/foo?label=bar`,
+        `${APIBase}/api/v1/namespaces/default/pods/foo?label=bar`,
+      );
+    });
+  });
+
+  describe("watchResource", () => {
+    it("should open a socket", async () => {
+      const socket = Kube.watchResource("v1", "pods", "default", "foo");
+      expect(socket.url).toBe(
+        `${WebSocketAPIBase}${APIBase}/api/v1/namespaces/default/pods?watch=true&fieldSelector=metadata.name%3Dfoo`,
       );
+      // it's a mock socket, so doesn't actually need to be closed
+      socket.close();
```

Comment:
where are you mocking WebSocket? I cannot find that

Reply:
It is mocked in `setupTests`: https://github.com/kubeapps/kubeapps/blob/f21c9840458fb9eea8e1ac7c9b7d866206e99309/dashboard/src/setupTests.ts#L6. This is necessary because `WebSocket` won't be defined in the test environment without it. So all WebSockets must be mocks.